### PR TITLE
[MT-55]:(feature) Password support for share links

### PR DESCRIPTION
**Pull request by PixoDev**
Provide password enable/disable support for Drive share links for folders and files

### CODE REVIEW

File: src/components/modals/SharedLinkSettingsModal/SharedLinkSettingsModal.tsx [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+            tailwind('flex flex-row px-5 overflow-hidden'),
+            { height: saveActionsHeight, opacity: saveActionsOpacity },
+          ]}
+        >
+          <AppButton
+            style={tailwind('flex-1 mr-1')}
+            title={strings.buttons.dismiss}
+            type={'cancel'}
+            onPress={function (): void {
+              throw new Error('Function not implemented.');
```
Comment (sg-gs) [inline src/components/modals/SharedLinkSettingsModal/SharedLinkSettingsModal.tsx:301]: why is this here?

Comment (PixoDev) [reply src/components/modals/SharedLinkSettingsModal/SharedLinkSettingsModal.tsx]: Fixed

File: src/components/modals/SharedLinkSettingsModal/SharedLinkSettingsModal.tsx [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+    }
+    setShareLinkPassword(newPassword);
+  };
+
+  const handleCopyLinkPress = async () => {
+    if (protectWithPassword && !shareLinkPassword) {
+      // You ned to provide a password
+      setPasswordError(true);
+      return;
+    } else {
```
Comment (sg-gs) [inline src/components/modals/SharedLinkSettingsModal/SharedLinkSettingsModal.tsx:135]: else is not required here, you are already doing a `return` on the if

File: src/components/modals/SharedLinkSettingsModal/SharedLinkSettingsModal.tsx [hunk 3]
```diff
@@ -0,0 +1,317 @@
...
+      setPasswordError(false);
+    }
+    if (generatedShareLink) {
+      Clipboard.setString(generatedShareLink);
```
Comment (sg-gs) [inline src/components/modals/SharedLinkSettingsModal/SharedLinkSettingsModal.tsx:139]: use a `return` here to remove the nesting which adds cognitive complexity.

File: src/components/modals/SharedLinkSettingsModal/SharedLinkSettingsModal.tsx [hunk 4]
```diff
@@ -0,0 +1,317 @@
...
+  };
+
+  const handleSaveShareLinkChanges = async () => {
+    if (!focusedShareItem) return;
+    if (!item) return;
+    if (protectWithPassword && !shareLinkPassword) {
+      // You ned to provide a password
+      setPasswordError(true);
+      return;
+    } else {
```
Comment (sg-gs) [inline src/components/modals/SharedLinkSettingsModal/SharedLinkSettingsModal.tsx:177]: `else` is not required here

**Review by sg-gs — CHANGES_REQUESTED** [review]

**Review by sg-gs — APPROVED** [review]

**Pull request merged**